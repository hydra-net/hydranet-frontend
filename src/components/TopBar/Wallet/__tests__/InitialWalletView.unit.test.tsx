import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { render } from "../../../../testUtils";
import InitialWalletView from "../InitialWalletView";

describe("<InitialWalletView/>", () => {
  let container: HTMLElement;
  const onClose = jest.fn();

  beforeEach(() => {
    const component = render(<InitialWalletView onClose={onClose} />);
    container = component.container;
  });

  it("Should render component", () => {
    expect(container).toMatchSnapshot();
  });

  it("Has a close button that calls onClose", () => {
    const closeBtn = screen.getByLabelText("close wallet");
    userEvent.click(closeBtn);
    expect(onClose).toHaveBeenCalled();
  });

  it("Has disconnect button", () => {
    const disconnectButton = screen.getByText(/disconnect/i);
    expect(disconnectButton).toBeInTheDocument();
  });
});
