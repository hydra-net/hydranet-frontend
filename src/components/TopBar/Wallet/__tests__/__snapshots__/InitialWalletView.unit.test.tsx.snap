// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InitialWalletView/> Should render component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation0 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        <div
          class="MuiBox-root MuiBox-root-3"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            style="line-height: 1.1; font-weight: 600; font-size: 0.975rem;"
          >
            MY WALLET
          </p>
        </div>
        <button
          aria-label="close wallet"
          class="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-4 MuiIconButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 20 20"
            >
              <path
                d="M3.56689 15.1035C3.21826 15.4521 3.20166 16.0747 3.5752 16.4399C3.94043 16.8052 4.56299 16.7969 4.91162 16.4482L9.9917 11.3599L15.0801 16.4482C15.437 16.8052 16.0513 16.8052 16.4165 16.4399C16.7734 16.0664 16.7817 15.4605 16.4165 15.1035L11.3364 10.0151L16.4165 4.93506C16.7817 4.57813 16.7817 3.96387 16.4165 3.59863C16.043 3.2417 15.437 3.2334 15.0801 3.59033L9.9917 8.67871L4.91162 3.59033C4.56299 3.2417 3.93213 3.2251 3.5752 3.59863C3.20996 3.96387 3.21826 4.58643 3.56689 4.93506L8.65527 10.0151L3.56689 15.1035Z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="MuiBox-root MuiBox-root-6"
        style="gap: 8px;"
      >
        <div
          class="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-7 MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
        >
          <div
            aria-disabled="false"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-9"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-10"
            >
              <div
                class="MuiBox-root MuiBox-root-12"
              >
                <img
                  height="28px"
                  src="hdx.svg"
                  style="margin-right: 8px;"
                  width="28px"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  HDX
                </p>
              </div>
              <div
                class="MuiBox-root MuiBox-root-13"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                  data-testid="balance-token"
                  style="font-weight: 600;"
                >
                  0.000000000
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-hidden"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <div
                  role="region"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-7 MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
        >
          <div
            aria-disabled="false"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-9"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-10"
            >
              <div
                class="MuiBox-root MuiBox-root-14"
              >
                <img
                  height="28px"
                  src="shdx.svg"
                  style="margin-right: 8px;"
                  width="28px"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  sHDX
                </p>
              </div>
              <div
                class="MuiBox-root MuiBox-root-15"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                  data-testid="balance-token"
                  style="font-weight: 600;"
                >
                  0.000000000
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-hidden"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <div
                  role="region"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-7 MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
        >
          <div
            aria-disabled="false"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-9"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-10"
            >
              <div
                class="MuiBox-root MuiBox-root-16"
              >
                <img
                  height="28px"
                  src="ghdx.svg"
                  style="margin-right: 8px;"
                  width="28px"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  gHDX
                </p>
              </div>
              <div
                class="MuiBox-root MuiBox-root-17"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2"
                  data-testid="balance-token"
                  style="font-weight: 600;"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                    style="width: 50px;"
                  />
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-hidden"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <div
                  role="region"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-18"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Disconnect
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
